src/back/glsl.rs

```diff
@@ -251,9 +251,43 @@ pub fn write<'a>(module: &'a Module, out: &mut impl Write, options: Options) ->
             })?
         };
 
-        if let Some(ref binding) = global.binding {
-            if !es {
-                write!(out, "layout({}) ", Binding(binding))?;
+        let storage_format = match module.types[global.ty].inner {
+            TypeInner::Image {
+                class: ImageClass::Storage(format, _),
+                ..
+            } => Some(format),
+            _ => None,
+        };
+
+        if global.binding.is_some() {
+            write!(out, "layout(")?;
+
+            if let Some(ref binding) = global.binding {
+                if !es {
+                    write!(out, "{}", Binding(binding))?;
+                }
+            }
+
+            if storage_format.is_some() && global.binding.is_some() {
```

Comment:
This is already inside an `if global.binding.is_some()` block. No need to check twice.